src/OrleansCodeGenerator/RoslynCodeGenerator.cs

```diff
@@ -445,8 +445,6 @@ private static GeneratedSyntax GenerateForAssemblies(List<Assembly> assemblies,
 
         private static void RecordType(Type type, Module module, Assembly targetAssembly, ISet<Type> includedTypes)
         {
-            if (type.IsNested) return;
```

Comment:
Thanks for the heads up - this is supposed to be merged after that PR because the test in this one require that nested types work (the test I expanded for my purposes was silently using the fallback serializer :disappointed:)